### ISSUE TITLE
Mobile WASM not able to load bevy website example

### ISSUE THREAD

**Issue by miketwenty1** [bevyengine/bevy#7038]
## Mobile seems to be sort of broken. At least for brave/chrome on android. 

Reopening since there is traction here to find a fix it seems & someone else also came across this:
https://github.com/bevyengine/bevy/issues/7038#issuecomment-1773104845

I'm unsure if this is a "known" or an actual issue. I found this when trying to do some WASM stuff of my own, then realized the website examples also won't load for my mobile device.
 I'm unable to reproduce this with virtualized chrome devices on desktop (need to use my actual phone). Firefox seems to be working in some cases. Chrome seems to work in less cases.

UiButton is an example where it works with firefox but not chrome. I encourage you to try this on your phone and see if it works. (let me know).
https://bevyengine.org/examples/ui/button/

- screenshots
(version 108.1.0 Build #2015920443)
![firefox](https://user-images.githubusercontent.com/7761473/209638576-5623a695-0eaa-47c0-bbcb-8ad4f91268f2.jpg)

(verison: 108.0.5359.128)
![chorme](https://user-images.githubusercontent.com/7761473/209638586-7806f49f-5c96-4b1c-a862-192c6363f1ce.jpg)

**Comment by nicopap** [comment]
I can get the examples working on my GraphenOS (Android 13) Pixel 6a on Vanadium 108.0 (a Chromium fork of same version) The button shows up, but tapping it seems to result in inconsistent behavior.

Hmm, you seem to have Chromium version 108.1, maybe something that broke in the latest version?

**Comment by mockersf** [comment]
Is it only something you see on UI examples? They are using some custom settings to only be active when there is user input, meaning you would need to be active in the canvas for them to load. I tried on an android device, it's harder and took longer on Chrome than in Firefox

**Comment by nicopap** [comment]
Ooh. So the button in the button example on Android only shows up after I tapped once on the screen (though the background has the default bevy bg, rather than dark-blue as on the screenshot).

**Comment by miketwenty1** [comment]
Tried multiple devices and it works. This is very odd, I'm only able to reproduce this on Pixel 6 Pro with chromium browsers (brave/chrome).
Even the Pixel 6 (non-pro) works!? with the ~same build version / version of chrome app. Note: firefox works fine for pixel 6 pro.

Since this isn't an overall android issue, and it may be something specific with Pixel 6 Pro chrome build, I'm going to close this.

**Comment by MeoMix** [comment]
Just noting that this is still an issue. My Pixel 6 Pro fails to load all Bevy WASM examples, but Firefox works fine. If there's any value in debug data here please let me know what to do. :)

**Comment by miketwenty1** [comment]
@MeoMix When i was debugging it, it was because of a screen size error.. maybe because the pixel pro has a curved screen or something and it doesn't interpret it correctly.. I didn't understand at the time an easy solution to fix it. There's probably some javascript magic you can do to override the bad values.

**Comment by MeoMix** [comment]
@miketwenty1 Fascinating. Alright, thanks for the heads up.

**Comment by miketwenty1** [comment]
@MeoMix if you don't mind.. if you do find find a fix, please update with findings.. I would really like to also implement it in my code base.

**Comment by miketwenty1** [comment]
reopening to track with you.

**Comment by MeoMix** [comment]
I think we need to sit tight for a while.

https://github.com/rust-windowing/winit/pull/2849 
https://github.com/rust-windowing/winit/issues/2524

winit has fixed the issue already but Bevy isn't using that version of winit. I don't think it's going to make it into Bevy 0.12. So my guess is that it'll work in 0.13, but that one could cherry-pick the change and patch their winit dependency locally if they were so inclined.

**Comment by miketwenty1** [comment]
@MeoMix both of those are now closed/merged. What do you think the current blocker is for this?

**Comment by MeoMix** [comment]
There is none AFAIK. winit 0.29 was merged into bevy main a couple of weeks ago. I haven't pulled down main to see if it fixed the issue for me yet though

**Comment by miketwenty1** [comment]
Yes this is still an issue while using main. The error is the same but reflects that it's using wgpu version 18 now.

```
.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.18.0/src/backend/direct.rs:3111:5:
wgpu error: Validation Error

Caused by:
    In Device::create_texture
      note: label = `main_texture_a`
    Dimension X value 4480 exceeds the limit of 4096

```

Is there a way I can override fix this?

**Comment by MeoMix** [comment]
Dang. I don't have any more information unfortunately. I was pretty sure that was the fix that needed to make it in, but, if it's not, it'll require triaging again and figuring out what else could be the issue.

**Comment by miketwenty1** [comment]
Is there a way to just override it so it doesn't exceed limits?

**Comment by MeoMix** [comment]
If I knew I would tell you. I only have about an hour of experience with this issue back when I triaged it the first time. I have no idea without spending time digging into it.

**Comment by miketwenty1** [comment]
I'm going to try a suggestion put here, i'll report back if this work around "works".
https://github.com/bevyengine/bevy/issues/7591#issuecomment-1705759085

**Comment by daxpedda** [comment]
@miketwenty1 which OS and browser are you using?
Would it be possible to run the following JS code on that device:
```js
const observer = new ResizeObserver((entries) => console.log(entries[0].devicePixelContentBoxSize[0]))
observer.observe(document.querySelector('canvas'))
```

**Comment by miketwenty1** [comment]
After a loading the game, I'm running the commands via USB Debugger:

<img width="719" alt="Screenshot 2024-01-14 at 9 55 26 AM" src="https://github.com/bevyengine/bevy/assets/7761473/dba4da0c-bff1-4c1c-8b71-2c84490d7fc0">

`ResizeObserverSize {inlineSize: 1442, blockSize: 2807}`

**Comment by daxpedda** [comment]
Okay, that looks about right to me. This is the size Winit should be reporting.
If Bevy is using a different size I assume it must not be coming from Winit.

> @miketwenty1 which OS and browser are you using?

That would still be helpful.

**Comment by miketwenty1** [comment]
these don't work.

chrome and chromium
brave v1.61.116
chrome v120.0.6099.210
duckduckgo v5.183.0

all on the pixel 6 pro OS: 
android v14

note:
firefox does work fwiw.

**Comment by daxpedda** [comment]
The resolution looks correct for this device.
I guess at this point somebody just has to dig into the code and figure out where Bevy is getting these much bigger numbers from.

**Comment by gnargfu** [comment]
Stumbled onto this issue myself when trying to run it in WebGL on a mobile device. Dug around a bit in the code and noticed that 
1. None of the fullscreen modes takes seems to take resolution into consideration this made it run on a PC (which was our previous target) but not on a mobile device. (of course and with the exception of `WindowMode::SizedFullscreen`)
2. `fit_canvas_to_parent` was set to true and that made the windowing take device ratio into consideration. 

My simple fix was to set it to `WindowMode::Windowed` and `fit_parent_to_canvas` to false. Also added CSS rules for maximizing the canvas to fill the parent. (This was 0.12, in 0.13 just ignore everything I commented in regards to `fit_parent_to_canvas`).
